File: Python3/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
import math

def solution(number, limit, power):
    aset, answer = set(), 0
    for i in range(1, number + 1):
        aset.clear()
        for j in range(1, int(math.sqrt(i)) + 1):
            if i % j == 0:
                aset.add(j)
                aset.add(i // j)
        # print(i,"의 약수는 ", aset)
        answer += (len(aset) if len(aset) <= limit else power)
    
    return answer

File: C++/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <vector>
#include <unordered_set>
#include <cmath>

using namespace std;

int solution(int number, int limit, int power) {
    int answer = 0;
    unordered_set<int> aset;
    
    for(int i = 1; i <= number; i++) {
        aset.clear();
        for(int j = 1; j <= sqrt(i); j++) {
            if(i % j == 0) {
                aset.insert(j);
                aset.insert(i / j);
            }
        }
        
        answer += aset.size() <= limit ? aset.size() : power;
    }
    
    return answer;
}